### ISSUE TITLE
Importing Custom Function Library and Description

### ISSUE THREAD

**Issue by madig500** [microsoft/pylance-release#4076]
Hello,

We are looking to  import a custom Library with its Definitions and Descriptions into Pylance. 

What are the steps to follow to include the Library and that the new functions with descriptions are available with Pylance?    Description of the functions of the library are in a Mark Down File.

regards,

Rolando

**Comment by rchiodo** [comment]
I'm assuming the library is written in Python?

If so, then Pylance will pick up its definitions automatically. 

Descriptions can be done with doc style comments:

```python
def foo():
   '''
   This is a function
   '''
```

Either in the source or in type stubs. For more information see the documentation here: https://microsoft.github.io/pyright/#/typed-libraries?id=docstrings

**Comment by rchiodo** [comment]
If the library is not written in python but has a python interface (like a pyc file), then you can still generate type stubs for the library. Pylance would then pick up the type stubs. 

See this documentation on how to create type stubs:
https://typing.readthedocs.io/en/latest/source/stubs.html

**Comment by madig500** [comment]
> 

HI,

Yes.   Functions are in Python  .py files ,  but descriptions of each functions are in a Mark Down file,  separate from the definition.   Basically we want to know how the reference is made between both in Pylance to retrieve it.    
Is it using docStrings?  


We are receiving the following message when Hover over our custom  functions

"[FunctionName]" is not defined Pylance(ReportUndefinedVariable)


Rolando

**Comment by rchiodo** [comment]
Yes, doc strings are how documentation is added to python files. 

Either directly in the python file itself or through type stubs. 

You'd use type stubs if you didn't want to put any other type information directly into the py files.

I believe doc strings were finalized in this pep:
https://peps.python.org/pep-0257/

**Comment by rchiodo** [comment]
There is no facility to have the doc strings in a separate file.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
